src/main/webapp/app/exam/manage/student-exams/student-exam-timeline/student-exam-timeline.component.ts

```diff
@@ -0,0 +1,388 @@
+        if (!submissionVersion) {
+            const programmingSubmission = this.programmingSubmissions.find((submission) => submission.submissionDate?.isSame(this.submissionTimeStamps[0]));
+            if (!programmingSubmission) {
+                return this.fileUploadSubmissions.find((submission) => submission.submissionDate?.isSame(this.submissionTimeStamps[0]));
+            } else {
+                return programmingSubmission;
+            }
+        }
+        return submissionVersion;
+    }
+
+    initializeExercise(exercise: Exercise, submission: Submission | SubmissionVersion | undefined) {
+        this.activeExamPage.exercise = exercise;
+        // set current exercise Index
+        this.exerciseIndex = this.studentExam.exercises!.findIndex((exercise1) => exercise1.id === exercise.id);
+        this.activateActiveComponent();
+        this.currentExercise = exercise;
+        this.currentSubmission = submission;
+        this.updateSubmissionOrSubmissionVersionInView();
+    }
+
+    private updateSubmissionOrSubmissionVersionInView() {
+        if (this.currentExercise?.type === ExerciseType.PROGRAMMING) {
+            this.updateProgrammingExerciseView();
+        } else if (this.currentExercise?.type === ExerciseType.FILE_UPLOAD) {
+            this.updateFileUploadExerciseView();
+        } else {
+            this.activePageComponent?.setSubmissionVersion(this.currentSubmission as SubmissionVersion);
+        }
+    }
+
+    private updateFileUploadExerciseView() {
+        const fileUploadComponent = this.activePageComponent as FileUploadExamSubmissionComponent;
+        if (fileUploadComponent) {
+            fileUploadComponent.studentSubmission = this.currentSubmission as FileUploadSubmission;
+            fileUploadComponent.updateViewFromSubmission();
+        }
+    }
+
+    private activateActiveComponent() {
+        this.pageComponentVisited[this.activePageIndex] = true;
+        const activeComponent = this.activePageComponent;
+        if (activeComponent) {
+            activeComponent.onActivate();
+        }
+    }
+
+    get activePageIndex(): number {
+        return this.studentExam.exercises!.findIndex((examExercise) => examExercise.id === this.activeExamPage.exercise?.id);
+    }
+
+    get activePageComponent(): ExamSubmissionComponent | undefined {
+        // we have to find the current component based on the activeExercise because the queryList might not be full yet (e.g. only 2 of 5 components initialized)
+        return this.currentPageComponents.find((submissionComponent) => (submissionComponent as ExamSubmissionComponent).getExercise().id === this.activeExamPage.exercise?.id);
+    }
+
+    /**
+     * This method is called when the user clicks on the slider
+     * @param changeContext the change context of the slider
+     */
+    onInputChange(changeContext: ChangeContext) {
```

Comment:
```suggestion
    onSliderInputChange(changeContext: ChangeContext) {
```
to make the intention clear in the name directly as well